generateBins.py

```diff
@@ -25,69 +13,46 @@
 
 
 # TODO Still need a good way of supplying these
-PFPR_FILE       = "rwa_pfpr2to10.asc"
+PFPR_FILE = "rwa_pfpr2to10.asc"
 POPULATION_FILE = "rwa_population.asc"
 
-
 # TODO Determine the bins computationally
 # Following bins are for Rwanda
 POPULATION_BINS = [2125, 5640, 8989, 12108, 15577, 20289, 27629, 49378, 95262, 286928]
 
-# getting data ready for binning
-# data should be 1-dimensional array, python list or iterable
-myArray  = np.loadtxt("GIS\\rwa_population.asc", skiprows = 6)
-print (type(myArray))
-print(myArray.ndim)
-print(myArray.shape)
-print(myArray.size)
-array_1d = myArray.flatten()
-print (array_1d)
-print(array_1d.ndim)
-print(array_1d.shape)
-print(array_1d.size)
-#gvf = 0.0
-#nclasses = 20
-
-def goodness_of_variance_fit(array, classes):
-    # get the break points
-    classes = jenkspy.jenks_breaks(array, classes)
-
-    # do the actual classification
-    classified = np.array([classify(i, classes) for i in array])
-
-    # max value of zones
-    maxz = max(classified)
-
-    # nested list of zone indices
-    zone_indices = [[idx for idx, val in enumerate(classified) if zone + 1 == val] for zone in range(maxz)]
-
-    # sum of squared deviations from array mean
-    sdam = np.sum((array - array.mean()) ** 2)
-
-    # sorted polygon stats
-    array_sort = [np.array([array[index] for index in zone]) for zone in zone_indices]
-
-    # sum of squared deviations of class means
-    sdcm = sum([np.sum((classified - classified.mean()) ** 2) for classified in array_sort])
-
-    # goodness of variance fit
-    gvf = (sdam - sdcm) / sdam
-
-    return gvf
 
-
-def process(configuration, gisPath = ""):
+def process(configuration, gisPath=""):
     # Load the configuration
     cfg = load_configuration(configuration)
+    filename = os.path.join(gisPath, PFPR_FILE)
 
     # TODO Add the stuff for the population bins!
+    # GVF Implementation
+    binning = bin_asc(filename)
+    #gvf = goodness_of_variance_fit(binning, 14)
+    #print("The value of Goodness of Variance fit is:", gvf)
+    #if gvf < 0.7:
+    #    print("Warning: GVF too low")
+
+    list_one = []
+    list_two = []
+    for x in range(5, 26):
```

Comment:
It's not too bad in this instance, but it's good practice to avoid raw numbers and use constants instead. When they are used, you should have at least a comment indicating why this particular range.

Reply:
as these values are used only once in the whole code, so I haven't defined constants.